mlflow/sklearn/__init__.py

```diff
@@ -994,3 +986,79 @@ def patched_fit(original, self, *args, **kwargs):
                 safe_patch(
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
+
+
+def eval_and_log_metrics(*, model, X, y_true, prefix, sample_weight=None):
```

Comment:
Is it a bit inconvenient if `model`, `X` and `y` are keyword-only arguments?

Reply:
Personally I find keyword arguments more robust in terms of avoiding mistakes but I am happy to go with the conventions of MLflow. Should we drop them?

Reply:
> I am happy to go with the conventions of MLflow.

Currently, there is no specific rule on when to use positional / keyword-only arguments in MLflow, but I'd prefer something similar to sklearn's convention (e.g. [cross_val_score](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html) where we can pass the first three arguments in positional form).

@smurching @mengxr What do you think?

Reply:
I think "model, X, y" order should be fine and they match the actions listed in the method name. The only question is whether we put `prefix` before or after `*`. Slightly prefer putting it after `*` because it is not obvious from the method name:

~~~
eval_and_log_metrics(model, X, y, prefix="val_")
~~~

Reply:
I went with mengxr's suggestion in the interest of time. If smurching has a different opinion then I can revert/change back to another scheme.

Reply:
+1 for making `prefix` keyword-only :)